src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NestedIfDepthCheck.java

```diff
@@ -42,6 +42,13 @@
  * <pre>
  * &lt;module name=&quot;NestedIfDepth&quot;/&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * if (x > 2) {
+ *     if (x > 3) {} // OK till here
+ *     else {} // violation, depth exceeded '1'
+ * }
```

Comment:
No violation, since depth is still '1' here.
Why?
because if-else is consider one unit, and does have similar depth. Moreover only one if-else block is nested under an if statement, so no violation.

Reply:
resolved